### DISCUSSION TITLE
Is there a way to make edges connect to the center of a node?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [xyflow/xyflow#2628] [Q&A] [answered]
Question: Is there a way to make edges connect to the center of a node?
For example, in the image below, I am rendering the same graph data with `reactflow` and the bottom panel is using another library which also has a similar API of passing in `nodes and `edges`.

![image](https://user-images.githubusercontent.com/6743796/205327379-70d44fe8-a98b-49f6-8bbe-ffd4a65053ec.png)

**Comment by bcakmakoglu** [comment]
You could probably use custom nodes that have "invisible" handles in the center of your node (just move the handle into the center of your node with CSS and hide it).
I don't have an example on hand but it shouldn't be too hard I believe.

**Reply by cliffordfajardo** [reply]
Thanks, I ended up creating a custom node for this 🙏

**Comment by cliffordfajardo** [comment, marked as answer]
### Solution
I created a tiny codesandbox here showing how to render the edge/handle at the center of the node
https://codesandbox.io/s/react-flow-edge-in-middle-2-way-node-x9m2nz?file=/src/App.tsx

This is what it looks like:
![CleanShot 2022-12-09 at 10 59 48@2x](https://user-images.githubusercontent.com/6743796/206776132-bd948d72-afbb-468d-942f-cef575495b2d.png)

### How it works:
1. Created a Custom Node
2. Added a `<Handle/>` inside my `CustomNode` to the node with custom styles. It renders exactly at the center of my node with custom CSS

```tsx
<Handle
    position={Position.Top}
    type="target"
    style={{
       background: "red",
       position: "absolute",
       top: "50%",
       right: "50%",
       transform: "translate(50%,-50%)"
    }}
/>
```
Created a Custom Edge

If you want the edge to appear behind the node, add a background color to your custom node, in my imag above, my CustomNode has no background color to it, so its transparent
